### PR TITLE
Yf add test empty string sam tag

### PR DESCRIPTION
**Pull request by yfarjoun**
### Description
Enables the reading and writing of sam/bam/cram records with empty string tags as required by https://github.com/samtools/hts-specs/issues/135

added tests to cover this functionality
### Checklist

- [x] Code compiles correctly
- [x] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by yfarjoun** [comment]
This is deep in the codec world where I've rarely been... @tfenne @alecw I think one of you should take a look....but if not, please help me identify a good candidate for this.

File: src/main/java/htsjdk/samtools/TextTagCodec.java [hunk 1]
```diff
@@ -149,12 +150,12 @@ public String encodeUntypedTag(final String tagName, final Object value) {
      */
     public Map.Entry<String, Object> decode(final String tag) {
         final int numFields = StringUtil.splitConcatenateExcessTokens(tag, fields, ':');
-        if (numFields != TextTagCodec.NUM_TAG_FIELDS) {
+        if (numFields != TextTagCodec.NUM_TAG_FIELDS && numFields != TextTagCodec.NUM_TAG_FIELDS - 1) {
```
Comment (alecw) [inline src/main/java/htsjdk/samtools/TextTagCodec.java]: Not a big deal, but might be simpler to test
if (numFields < TextTagCodec.NUM_TAG_FIELDS - 1)

Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/TextTagCodec.java]: I wanted to protect against a possibility of numFields>3, given that the original code protected against that.

Comment (alecw) [inline src/main/java/htsjdk/samtools/TextTagCodec.java]: Hi Yossi, that's not possible given the contract of StringUtil.splitConcatenateExcessTokens, but again, not a big deal.

**Comment by yfarjoun** [comment]
Thanks for looking.
